### PR TITLE
[CoreML] Update to Xcode 9 Beta 4

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/coreml.cs [hunk 1]
```diff
@@ -266,5 +276,46 @@ interface MLMultiArray {
 		[Export ("setObject:forKeyedSubscript:")]
 		void SetObject (NSNumber obj, NSNumber [] key);
 	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLDictionaryConstraint {
```
Comment (spouliot) [inline src/coreml.cs]: I don't see this type exposed anywhere in the PR !?! but it's used as a property in the diff. Something is missing ?

File: src/coreml.cs [hunk 2]
```diff
@@ -266,5 +276,46 @@ interface MLMultiArray {
...
+
+		[Export ("keyType")]
+		MLFeatureType KeyType { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLImageConstraint {
```
Comment (spouliot) [inline src/coreml.cs]: same

File: src/coreml.cs [hunk 3]
```diff
@@ -266,5 +276,46 @@ interface MLMultiArray {
...
+		[Export ("pixelsWide")]
+		nint PixelsWide { get; }
+
+		[Export ("pixelFormatType")]
+		uint PixelFormatType { get; }
+	}
+
+	[Watch (4,0), TV (11,0), Mac (10,13, onlyOn64: true), iOS (11,0)]
+	[BaseType (typeof (NSObject))]
+	interface MLMultiArrayConstraint {
```
Comment (spouliot) [inline src/coreml.cs]: same

**Comment by dalexsoto** [comment]
Yeah this makes sense because `MLDictionaryConstraint`, `MLImageConstraint` and `MLMultiArrayConstraint` have **getter* only properties. Will file a radar

**Comment by spouliot** [comment]
bind the properties as readonly
and add `[DisableDefaultCtor]` on the types
until we get an answer back :)

**Comment by dalexsoto** [comment]
Yup, just filled a rada, just doing final checks 👍

**Comment by dalexsoto** [comment]
@spouliot Done 👍

**Pull request merged**